### PR TITLE
Refactors and tweaks reactive armours

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
Reactive armours no longer block or get activated by ion shots.
Reactive armours now block tasers.
Pyro armour is now fully fire proof instead of giving you negative fire stacks when it hits.
Repulse armour makes sparkly pulses when it activates.
Removes a couple of spawns from the code.

## Why It's Good For The Game

Ions currently have the habit of disabling the armour, but then the armour procs teleporting them away/stunning/damaging you, so you cannot follow up on it, affectively neutralising this weakness 50% of the time if you hit them directly. 

Tasers should be blocked, they are dangerous to the user and doesn't make much sense for them not to be blocked.

Pyro armour is already *sorta* fire proof, but the way it is done is very janky. negative fire stacks is very clunky in my opinion, and just making the wearer immune to fire is a much better solution.

Repulse armour making sparkles is cool, and fits in with the theme of a force wave pushing you back.

Spawns are generally bad and should be removed

## Changelog
:cl:
add: reactive repulse armour now makes sparkles
tweak: Reactive armours now block tasers. They now don't block/get triggered by Ions.
tweak: incendiary armour is properly fireproof
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/armor.dm [hunk 1]
```diff
@@ -416,38 +421,43 @@
...
 					M.Weaken(3)
 					to_chat(M, "<span class='userdanger'>You're slammed into the floor by [owner]'s reactive armor!</span>")
 			else
+				new sparkle_path(get_turf(AM), get_dir(owner, AM))
 				if(isliving(AM))
 					var/mob/living/M = AM
 					to_chat(M, "<span class='userdanger'>You're thrown back by the [owner]'s reactive armor!</span>")
-				spawn(0)
-					AM.throw_at(throwtarget, ((clamp((repulse_power - (clamp(distfromuser - 2, 0, distfromuser))), 3, repulse_power))), 1)//So stuff gets tossed around at the same time.
+				INVOKE_ASYNC(src, .proc/repulse, AM, throw_target, repulse_power, dist_from_user)//So stuff gets tossed around at the same time.
```
Comment (SteelSlayer) [inline code/modules/clothing/suits/armor.dm:454]: I'm pretty sure you can skip the repulse proc and just do:
```dm
INVOKE_ASYNC(AM, .proc/throw_at, throw_target, ...) // calculations etc.
```

Comment (hal9000PR) [reply code/modules/clothing/suits/armor.dm:454]: oh yeah that makes a lot more sense

File: code/modules/clothing/suits/armor.dm [hunk 2]
```diff
@@ -387,10 +390,12 @@
 		E.Goto(owner, E.move_to_delay, E.minimum_distance)
 		owner.alpha = 0
 		owner.visible_message("<span class='danger'>[owner] is hit by [attack_text] in the chest!</span>") //We pretend to be hit, since blocking it would stop the message otherwise
-		spawn(40)
-			owner.alpha = initial(owner.alpha)
+		addtimer(CALLBACK(src, .proc/remove_invisible, owner), 4 SECONDS)
 		return TRUE
 
+/obj/item/clothing/suit/armor/reactive/stealth/proc/remove_invisible(mob/living/carbon/human/owner)
+	owner.alpha = initial(owner.alpha)
```
Comment (S34NW) [inline code/modules/clothing/suits/armor.dm:397]: How does this interact with a vampire that moves into darkness with shadow cloak on during the stealth period?

Comment (hal9000PR) [reply code/modules/clothing/suits/armor.dm:397]: it resets it for a brief moment, but then it goes back to being cloaked.

Comment (S34NW) [reply code/modules/clothing/suits/armor.dm:397]: Consider this resolved

**Review by dearmochi — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Review by Kyep — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**